Koans/AboutEnumerations.cs

```diff
@@ -0,0 +1,134 @@
+                  While the underlying type of the instance above will be an integer,
+                  enums are first class types in the C# language. The type of the
+                  variable 'dagobah' will be the enum itself.
+                  */
+                  Assert.Equal(typeof(FillMeIn), dagobah.GetType());
+
+                  /*
+                  Adding new members to an enum is straight-forward and as you'd expect.
+                  Try adding a new member to the 'Planet' enum called 'Tantooine'.
+                  */
+                  Assert.True(Enum.IsDefined(typeof(Planet), "Tantooine"));
+            }
+
+            [Step(2)]
+            public void CastingToAnEnum()
+            {
+                  /*
+                  Because the underlying type of enum members are actually integers,
+                  you can create instances of an enum by casting an integer value to
+                  an enum. 
+
+                  Note that the associated constant value of members start with zero
+                  and increase by one.
+                  */
+                  var forestPlanet = (Planet)1;
+                  Assert.Equal(FILL_ME_IN, forestPlanet);
+
+                  /*
+                  Why would casting integers to enums be valuable? You may want to
+                  save values to a database perhaps. Those values would be stored as
+                  an integer and when querying for those values, you'd be handed an
+                  integer back. For instance:
+
+                  var usersHomePlanet = (Planets)row['planet'];
+
+                  Handling integer values as enumerations improves readability.
+                  */
+            }
+
+            [Step(3)]
+            public void MemberValuesCanBeExplicit()
+            {
+                  /*
+                  There may be cases where you don't want enum members to start at zero
+                  and incrementing by one. Perhaps you want them incrementing by 100.
+                  */
+                  var logLevel = LogLevel.Verbose;
+                  Assert.Equal(300, (int)logLevel);
+            }
+
+            [Step(4)]
+            public void EnumsCanBeFlags()
+            {
+                  /*
+                  Enums can represent a combination of choices. When an enum is
+                  decorated with the '[Flag]' attribute, it changes the underlying
+                  type to a bit field. This allows for multiple members to be assigned
+                  to a single value.
+
+                  Notice we're using the OR operator to concatenate multiple days.
+
+                  We're missing Friday!
+                  */
+
+                  var workWeek = DayOfTheWeek.Monday | DayOfTheWeek.Tuesday | DayOfTheWeek.Wednesday | DayOfTheWeek.Thursday;
+
+                  Assert.True(workWeek.HasFlag(DayOfTheWeek.Monday));
+                  Assert.True(workWeek.HasFlag(DayOfTheWeek.Friday)); // Assuming you work Fridays :)
```

Comment:
Will these asserts fail? Remember we want the learner to figure out how to make them pass.

Reply:
The first one will pass without doing anything. My intention was to show what works and resolve the failing one, but I agree, why put an assert in that passes. I'll remove the first.

Reply:
Maybe use Assert.AreEqual so that the FILL_ME_IN object can be used.

Reply:
I this particular case, I wanted to use `workWeek.HasFlag` to make sure the student understands how to OR in an additional enum value. I won't be able to use that with an `Assert.AreEqual`. I did however add `FILL_ME_IN' to the variable assignment, so the student has some a pointer to the solution.

Reply:
Does this compile? Friday does not exist in the flag set.

Reply:
It compiles, but the assert will fail because the `HasFlag` check returns false, Friday is not in the `workWeek` variable.